### ISSUE TITLE
Fine tuning the model

### ISSUE THREAD

**Issue by tamanna-sb** [tamanna037/InformationTypesDetectionNLP#1]
The hyperparameter of the model needs to be fine tuned for training. 

**HyperParameter Tuning**
Hyperparameters are the knobs or settings that can be tuned before running a training job to control the behaviour of an ML algorithm. They can have a big impact on model training as it relates to training time, infrastructure resource requirements (and as a result cost), model convergence and model accuracy. To get good results from Machine Learning (ML) models, it is necessary to always tune hyperparameters—learning rate, regularization, etc. This tuning is critical for performance and accuracy. 

**Hyperparameters to tune:**

1. Batch Size — This value indicates number of training samples that are used within each epoch. With smaller batch sizes it is observed that it is easy to get out of “[local minima](https://en.wikipedia.org/wiki/Maxima_and_minima)”. However, larger batch size can end up getting stuck in the wrong solution.
2. Learning rate — determines the step size at each iteration (epoch) while moving toward a minimum of a loss function. It is crucial to achieve right learning rate as a larger learning rate would mean that model might overshoot the optimal solution and a smaller learning rate will increase training time to find the optimal solution.

**2 practical tips  to use for BERT fine tuning**
1. Evaluate the model multiple times during an epoch; and
2. Identify bad initializations early and stop them.
You can go through this [article](https://towardsdatascience.com/how-to-make-the-most-out-of-bert-finetuning-d7c9f2ca806c) for more insights.

Use validation data for evaluation.

**Comment by tamanna-sb** [comment]
Trainingargs has been set in InfoTypesDetectionOss.py file ( [10](https://github.com/tamanna037/InformationTypesDetectionNLP/pull/10) )


<img width="389" alt="Screen Shot 2022-10-26 at 12 21 56 PM" src="https://user-images.githubusercontent.com/14105107/198106158-39a17483-2d1c-4e47-a775-84144f617f85.png">
